.changes/next-release/Feature-eae23041-03ad-4df1-92ab-908412b5f584.json

```diff
@@ -0,0 +1,4 @@
+{
+	"type": "Feature",
+	"description": "Auto-connect tries a maximum of one non-default profile instead of three"
```

Comment:
IMO auto-connect should only try to use whatever is saved in `profile`, nothing more.
And for Cloud9 profile should be `default` by default.

The current credentials system can't easily support it but there should be 3 states for auth:
* Fully logged-out
* Logged-in but disconnected
* Logged-in and connected

Failed auto-connects should go into the second state. Of course, this is out-of-scope for this PR.

Reply:
> And for Cloud9 profile should be default by default.

If `default` doesn't exist for some reason, why not use IMDS / env vars?

Reply:
> If default doesn't exist for some reason, why not use IMDS / env vars? It is powerful for Toolkit to work out-of-the-box in as many places as possible (EKS, EC2, ...).

I guess picking a profile in the absence of an explicit user profile is fine as long as it's clear about what the Toolkit is doing. I'm more against auto-connecting random profiles when the user already manually picked one. Having the auto-connect is nice for getting started but beyond that it can be annoying.